src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -396,12 +396,12 @@ internal List<Message> DequeueAllWaitingRequests()
 
         public void DelayDeactivation(TimeSpan timespan)
         {
-            if (timespan <= TimeSpan.Zero)
+            if (timespan <= TimeSpan.Zero && timespan != Timeout.InfiniteTimeSpan)
```

Comment:
The code could be a little simplified by swapping this with the `else if`, so that `InfiniteTimeSpan` only needs to be checked once.

Reply:
changed in https://github.com/dotnet/orleans/pull/8712